### PR TITLE
Fix dev env on aarch64 hosts

### LINKED ISSUE #1375: Setup local env fails
Hi,

I'm trying to setup a local environment to develop a custom policy, but I fail to get APIcast up and running as described...

##### Version

`master` branch

##### Steps To Reproduce

1. `git clone git@github.com:3scale/APIcast.git`
2. `cd APIcast`
3. `make development`
4. `make dependencies` --> fails

##### Current Result

```
domi 🙈 APIcast $ make development
Running on Darwin
docker compose -f docker-compose-devel.yml -f docker-compose-devel-volmount-mac.yml up -d
[+] Running 2/2
 ⠿ Container apicast_build_0-redis-1        Running                                                                            0.0s
 ⠿ Container apicast_build_0-development-1  Started                                                                            1.5s
docker compose -f docker-compose-devel.yml -f docker-compose-devel-volmount-mac.yml exec -e COLUMNS="`tput cols`" -e LINES="`tput lines`" --user 501:20 development bash
bash-4.4$ make dependencies
/usr/local/openresty/luajit/bin/rover lock --roverfile=/opt/app-root/src/gateway/Roverfile
Resolving /opt/app-root/src/gateway/Roverfile
Cloning into 'lua-resty-repl'...
fatal: unable to connect to github.com:
github.com[0: 140.82.121.3]: errno=Connection refused

/usr/local/openresty/luajit/bin/luajit-2.1.0-beta3: /usr/share/lua/5.1/rover/lock.lua:169: attempt to index local 'rockspec' (a nil value)
stack traceback:
        /usr/share/lua/5.1/rover/lock.lua:169: in function 'expand_dependencies'
        /usr/share/lua/5.1/rover/lock.lua:199: in function 'resolve'
        /usr/share/lua/5.1/rover/cli/lock.lua:23: in function </usr/share/lua/5.1/rover/cli/lock.lua:11>
        /usr/share/lua/5.1/rover/cli.lua:33: in function </usr/share/lua/5.1/rover/cli.lua:27>
        /usr/lib64/luarocks/rocks/lua-rover/scm-1/bin/rover:8: in main chunk
        [C]: at 0x55ad138087a0
make: *** [Makefile:214: /opt/app-root/src/gateway/Roverfile.lock] Error 1
bash-4.4$ ^C
bash-4.4$ exit
```

##### Additional Information

* I tried to access github.com from within the running container, it works with `git` (ssh and https) and `curl` (https)

### PR DESCRIPTION
**Pull request by guicassolato**
Fixes multiple issues to build and run the development environment on aarch64-based host machines, caused by several incompatible dependencies.

- [x] build and run dev image targeting linux/amd64 platform
- [x] deps: [IPC::Run](https://metacpan.org/pod/IPC::Run) (preinstalled) - because otherwise Test-Nginx cannot be installed
- [x] deps: [lua-resty-jwt](https://luarocks.org/modules/cdbattags/lua-resty-jwt) v0.2.0 (installed with Rover) - because latest version (v0.2.3) [requires](https://github.com/cdbattags/lua-resty-jwt/blob/b3d5c085643fa95099e72a609c57095802106ff9/lua-resty-jwt-dev-0.rockspec#L19) incompatible lua-resty-openssl
- [x] deps: [jsonschema](https://luarocks.org/modules/membphis/jsonschema) v0.8 (installed with Rover) - because latest version (v0.9.8) [requires](https://github.com/api7/jsonschema/blob/0096f5e5bfdd210136a98e9d50799d86b9bd040c/rockspec/jsonschema-0.9.8-0.rockspec#L23) incompatible lrexlib-pcre.
- [x] deps: [luaossl](https://luarocks.org/modules/daurnimator/luaossl) 20200709 (preinstalled) - because latest (20220711) cannot be compiled with OpenSSL 1.1.1 on an aarch64 host ([ref](https://github.com/wahern/luaossl/issues/175#issuecomment-1310378633))
- [x] deps: deps: [nginx-lua-prometheus](https://luarocks.org/modules/knyar/nginx-lua-prometheus) v0.20181120 - to avoid _"module 'nginx.prometheus' not found"_ when running the tests
- [x] deps: [lua-resty-http](https://luarocks.org/modules/pintsized/lua-resty-http) v0.15 - to avoid `string` vs `table` type when deep invoking `is_fqdn` from `apicast.configuration_loader.remote_v1:download → resty.resolver.http:request_uri`
- [x] fix: test assertion of a falsy value passing no argument – no longer the same as passing `nil` since [luassert](https://luarocks.org/modules/lunarmodules/luassert) v1.9.0.

Closes #1375.

### Verification steps

(Optional) Build your own dev image:

```sh
make dev-build
```

Run the dev env:

```sh
make development IMAGE=apicast-development:latest
```

...or using a prebuilt image containing the fixes:

```sh
make development IMAGE=quay.io/guicassolato/3scale:apicast-dev-20221124
```

Within the container, install the dependencies:

```sh
make dependencies
```

After several minutes, installation should finish successfully:

```
Complete! Modules were bundled into /opt/app-root/src/vendor/cache
```

Run the tests:

```sh
make busted
# 1208 successes / 0 failures / 0 errors / 3 pending : 24.479539 seconds
```

```sh
make prove
# All tests successful.
# Files=85, Tests=3397, 1601 wallclock secs ( 1.52 usr  0.20 sys + 3998.34 cusr 406.44 csys = 4406.50 CPU)
# Result: PASS
# Success
```

### Future improvement

Ditch Lua Rover altogether and use LuaRocks directly instead to install the dependencies.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,7 +72,7 @@ dev-build: export OPENRESTY_RPM_VERSION?=1.19.3
 dev-build: export LUAROCKS_VERSION?=2.3.0
 dev-build: IMAGE_NAME ?= apicast-development:latest
 dev-build: ## Build development image
-	$(DOCKER) build -t $(IMAGE_NAME) \
+	$(DOCKER) build --platform linux/amd64 -t $(IMAGE_NAME) \
```
Comment (eguzki) [inline Makefile:75]: Shouldn't be this multiplatform?

File: Dockerfile.devel [hunk 2]
```diff
@@ -12,7 +12,7 @@ ENV APP_ROOT=/opt/app-root \
 
 RUN sed -i s/enabled=./enabled=0/g /etc/yum/pluginconf.d/subscription-manager.conf
 
-RUN yum upgrade -y 
+RUN yum upgrade -y
```
Comment (eguzki) [inline Dockerfile.devel:15]: When there are changes in the devel image, a new image should be pushed to the registry and the [circle-ci config](https://github.com/3scale/APIcast/blob/a4f11234e56035d3c8d152a6f26e77a8093e6cee/.circleci/config.yml#L115) and [docker compose files](https://github.com/3scale/APIcast/blob/a4f11234e56035d3c8d152a6f26e77a8093e6cee/docker-compose-devel.yml#L5) should be updated.

Currently the image url+tag is `quay.io/3scale/apicast-ci:openresty-1.19.3`. `1.19.3` makes reference to the openresty version. I suggest a new tag

```
quay.io/3scale/apicast-ci:openresty-1.19.3-01
```

**Comment by eguzki** [comment]
Which `docker-compose` version are you using? 

With `docker-compose version 1.25.0, build unknown`, I get the follwing when running `make development IMAGE=apicast-development:latest`

```bash
❯ make development IMAGE=apicast-development:latest
Running on Linux
docker-compose -f docker-compose-devel.yml -f docker-compose-devel-volmount-default.yml up -d
ERROR: The Compose file './docker-compose-devel.yml' is invalid because:
Unsupported config option for services.development: 'platform'
make: [Makefile:204: development] Error 1 (ignored)
docker-compose -f docker-compose-devel.yml -f docker-compose-devel-volmount-default.yml exec -e COLUMNS="`tput cols`" -e LINES="`tput lines`" --user 1000:1000 development bash
ERROR: The Compose file './docker-compose-devel.yml' is invalid because:
Unsupported config option for services.development: 'platform'
make: *** [Makefile:206: development] Error 1
```

BTW, should we replace `docker-compose` python based v1 of https://github.com/docker/compose to v2, written in go?? (not part of this PR obviously)

**Comment by guicassolato** [comment]
> Which `docker-compose` version are you using?

```
$ docker-compose version
Docker Compose version v2.12.2
```

**Comment by eguzki** [comment]
working locally :woman_dancing: 

we can expect tests in CircleCI to fail as they are using old dev image.

**Comment by eguzki** [comment]
I have pushed the new dev image to quay: `quay.io/3scale/apicast-ci:openresty-1.19.3-pr1381`

Can you update the image URL references from the repo?? That should make the tests pass. The references:

```bash
❯ ag --hidden quay.io/3scale/apicast-ci:openresty-1.19.3
docker-compose-devel.yml
5:    image: ${IMAGE:-quay.io/3scale/apicast-ci:openresty-1.19.3}

Makefile
15:DEVEL_IMAGE ?= quay.io/3scale/apicast-ci:openresty-1.19.3
66:#     make dev-build IMAGE_NAME=quay.io/3scale/apicast-ci:openresty-1.19.3
68:#     docker push quay.io/3scale/apicast-ci:openresty-1.19.3

.circleci/config.yml
115:      - image: quay.io/3scale/apicast-ci:openresty-1.19.3
```

**Comment by eguzki** [comment]
awesome work @guicassolato :medal_military:

**Review by eguzki — APPROVED** [review]

**Pull request merged**